File: swift/cat.swift
let input = readLine()!
var c = 0, a = 0, t = 0

input.characters.forEach { (ch) in
    switch (ch) {
    case "c":
        c += 1
        break
    case "a":
        a += 1
        break
    case "t":
        t += 1
        break
    default:
        break
    }
}

let maxNum = max(c, a, t)
let minNum = min(c, a, t)

print(minNum)
print(maxNum - c)
print(maxNum - a)
print(maxNum - t)

File: cpp/cat.cpp
#include <iostream>
using namespace std;

string readline() {
    string str;
    getline(cin, str);
    return str;
}

int main(void){
    auto c = 0, a = 0, t = 0;
    const auto input = readline();
    for (auto ch : input) {
        switch (ch) {
        case 'c':
            c++;
            break;
        case 'a':
            a++;
            break;
        case 't':
            t++;
            break;
        default:
            break;
        }
    }
    auto maxNum = max({c, a, t});
    auto minNum = min({c, a, t});
    
    cout << minNum << endl;
    cout << maxNum - c << endl;
    cout << maxNum - a << endl;
    cout << maxNum - t << endl;

    return 0;
}

